rules/rules-reviewed/camel3/camel2/xml-317-changes.windup.xml

```diff
@@ -0,0 +1,294 @@
+                    <artifact groupId="org.apache.camel" artifactId="camel-sip" />
+                </project>
+            </when>
+            <perform>
+                <hint title="`camel-sip` component has been removed" effort="7" category-id="mandatory" >
+                    <message>`org.apache.camel:camel-sip` component has been removed in Camel 3.17.0.  </message>
+                    <link href="https://camel.apache.org/manual/camel-3x-upgrade-guide-3_17.html#_deprecated_components" title="Camel 3 - Migration Guide: Removed components" />
+                </hint>
+            </perform>
+        </rule>
+        <rule id="xml-removed-camel317-00015">
+            <when>
+                <project>
+                    <artifact groupId="org.apache.camel" artifactId="camel-soroush" />
+                </project>
+            </when>
+            <perform>
+                <hint title="`camel-soroush` component has been removed" effort="7" category-id="mandatory" >
+                    <message>`org.apache.camel:camel-soroush` component has been removed in Camel 3.17.0.  </message>
+                    <link href="https://camel.apache.org/manual/camel-3x-upgrade-guide-3_17.html#_deprecated_components" title="Camel 3 - Migration Guide: Removed components" />
+                </hint>
+            </perform>
+        </rule>
+
+        <rule id="xml-removed-camel317-00016">
+            <when>
+                <project>
+                    <artifact groupId="org.apache.camel" artifactId="camel-tagsoup" />
+                </project>
+            </when>
+            <perform>
+                <hint title="`camel-tagsoup` component has been removed" effort="7" category-id="mandatory" >
+                    <message>`org.apache.camel:camel-tagsoup` component has been removed in Camel 3.17.0.  </message>
+                    <link href="https://camel.apache.org/manual/camel-3x-upgrade-guide-3_17.html#_deprecated_components" title="Camel 3 - Migration Guide: Removed components" />
+                </hint>
+            </perform>
+        </rule>
+        <rule id="xml-removed-camel317-00017">
+            <when>
+                <project>
+                    <artifact groupId="org.apache.camel" artifactId="camel-yammer" />
+                </project>
+            </when>
+            <perform>
+                <hint title="`camel-yammer` component has been removed" effort="7" category-id="mandatory" >
+                    <message>`org.apache.camel:camel-yammer` component has been removed in Camel 3.17.0.  </message>
+                    <link href="https://camel.apache.org/manual/camel-3x-upgrade-guide-3_17.html#_deprecated_components" title="Camel 3 - Migration Guide: Removed components" />
+                </hint>
+            </perform>
+        </rule>
+
+        <rule id="xml-legacy-camel317-00001">
+            <when>
+                <javaclass references="org.apache.camel.builder.DeadLetterChannelBuilder" as="javaClass">
```

Comment:
The `javaclass` `as` attribute can be removed. Also in every rule where this variable is not being used. Take a look [here](https://access.redhat.com/documentation/en-us/migration_toolkit_for_applications/5.2/html-single/rules_development_guide/index#construct_a_javaclass_element) for more info on what this does.

Reply:
Removed